solvation_analysis/analysis_library.py

```diff
@@ -0,0 +1,65 @@
+import pandas as pd
+
+
+class _IonSpeciation:
+    """
+    A class for calculating and storing speciation information for solvents.
```

Comment:
See comments on simplifying inputs away from a dataframe.

That way also your `_IonSpeciation` function is modular and can be constructed from any set of numpy arrays, even if obtained outside of `Solute`. If they don't rely on a specific dataframe input they don't need to be (semi) private and `_IonSpeciation` can become  just `IonSpeciation`